packages/access-token-jwt/test/dpop-verifier.test.ts

```diff
@@ -0,0 +1,1108 @@
+      accessTokenClaims: {
+        cnf: { jkt: 'some-thumbprint' },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidRequestError);
+    await expect(result).rejects.toThrow(
+      'Invalid Authorization HTTP Header format'
+    );
+  });
+
+  it('throws if multiple DPoP headers are present', async () => {
+    headers.dpop = `${proof},${proof}`;
+    await expect(verifyDPoP(validOptions())).rejects.toThrow(
+      InvalidRequestError
+    );
+  });
+
+  it('throws if jti is missing', async () => {
+    const { privateKey } = await generateKeyPair('ES256');
+    const badProof = await new SignJWT({
+      htm: method,
+      htu: url,
+      iat: Math.floor(Date.now() / 1000),
+      ath: crypto.createHash('sha256').update(jwt).digest('base64url'),
+    })
+      .setProtectedHeader({ typ: 'dpop+jwt', alg: 'ES256', jwk })
+      .sign(privateKey);
+
+    headers.dpop = badProof;
+    await expect(verifyDPoP(validOptions())).rejects.toThrow(InvalidProofError);
+  });
+
+  it('throws if typ header is incorrect', async () => {
+    const { privateKey } = await generateKeyPair('ES256');
+    const badProof = await new SignJWT({
+      htm: method,
+      htu: url,
+      iat: Math.floor(Date.now() / 1000),
+      jti: crypto.randomUUID(),
+      ath: crypto.createHash('sha256').update(jwt).digest('base64url'),
+    })
+      .setProtectedHeader({ typ: 'wrong', alg: 'ES256', jwk })
+      .sign(privateKey);
+
+    headers.dpop = badProof;
+    await expect(verifyDPoP(validOptions())).rejects.toThrow(InvalidProofError);
+  });
+
+  it('throws InvalidProofError if iat is missing in proof', async () => {
+    const { publicKey, privateKey } = await generateKeyPair('ES256');
+
+    // Sign a valid DPoP proof with NO `iat`
+    const jwt = 'header.payload.signature';
+    const encodedAth = crypto
+      .createHash('sha256')
+      .update(jwt)
+      .digest('base64url');
+
+    const proof = await new SignJWT({
+      htm: 'POST',
+      htu: 'https://api.example.com/resource',
+      jti: crypto.randomUUID(),
+      ath: encodedAth,
+    })
+      .setProtectedHeader({
+        alg: 'ES256',
+        typ: 'dpop+jwt',
+        jwk: await exportJWK(publicKey),
+      })
+      .sign(privateKey);
+
+    const result = verifyDPoP({
+      jwt,
+      method: 'POST',
+      url: 'https://api.example.com/resource',
+      headers: {
+        authorization: `DPoP ${jwt}`,
+        dpop: proof,
+      },
+      supportedAlgorithms: ['ES256'],
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Missing "iat" claim in DPoP proof');
+  });
+
+  it('throws InvalidProofError if iat is outside the acceptable range', async () => {
```

Comment:
```suggestion
  it('throws InvalidProofError if iat is in the past', async () => {
```

There is another test case called `throws if iat is in the future`, so this way it's clearer this test case is about an iat in the past, and the other is about an iat in the future.

Reply:
Fixed here: https://github.com/auth0/node-oauth2-jwt-bearer/pull/183/commits/2f6096f0e67f7a7a7c04603153160a3b61655378